src/Pact/Repl/Lib.hs

```diff
@@ -301,23 +302,44 @@ continuePact i as = case as of
   where
     go :: Integer -> Bool -> Maybe Text -> Maybe (ObjectMap (Term Name)) -> Eval LibState (Term Name)
     go step rollback pid userResume = do
-      pactExec <- use evalPactExec
-      pactId <- case pid of
-        Nothing -> maybe
-                   (evalError' i "continue-pact: no pact exec in context")
-                   (return . _pePactId)
-                   pactExec
-        Just pidTxt -> return $ PactId pidTxt
-      let pactStep = PactStep
-                     (fromIntegral step)
-                     rollback pactId userResume
+      pe <- use evalPactExec
+      (pactId, y) <- unwrapExec pid userResume pe
+
+      let pactStep = PactStep (fromIntegral step) rollback pactId y
       viewLibState (view rlsPacts) >>= \pacts ->
         case M.lookup pactId pacts of
           Nothing -> evalError' i $ "Invalid pact id: " <> pretty pactId
           Just PactExec{..} -> do
             evalPactExec .= Nothing
             local (set eePactStep $ Just pactStep) $ resumePact (_faInfo i) Nothing
 
+    unwrapExec mp mo Nothing = do
+      pid <- case mp of
+        Nothing -> evalError' i
+          "continue-pact: No pact id supplied and no pact exec in context"
+        Just p -> pure $ PactId p
+      y <- case mo of
+        Nothing -> pure Nothing
+        Just o -> toYield i pid o
+
+      pure (pid, y)
+    unwrapExec mp mo (Just e) = do
+      let pid = maybe (_pePactId e) PactId mp
+      y <- case mo of
+        Nothing -> pure . _peYield $ e
+        Just o -> toYield i pid o
+      pure (pid, y)
+
+    -- This is only called in the case where no resume object
+    -- is supplied. The function converts the object map into
+    -- a map of pact values, and constructs the endorsement
+    -- for the yield with an empty chain id
+    toYield j p o = do
+      o' <- enforcePactValue' o
+      let t = ChainId ""
```

Comment:
Here, we're supplying an empty `ChainId` in the case where no previous yield is found. We'll need to discuss